tests/test_matrix.py

```diff
@@ -0,0 +1,271 @@
+import pytest
+
+from matrix.matrix import Matrix
+
+CONSTANT_NUMBER_LIST = [50, -2.5, 0, 2]
+LEGAL_MATRIX_LIST = [(()), ((1,),), ((1, 2), (-3, -4)), ((1, 2.5, 3), (3, 4, 5), (1, -5.5, 6))]
+ILLEGAL_TYPE_MATRIX_LIST = [None, 1, "test", [(1,)]]
+ILLEGAL_TYPE_MATRIX_CELL = [(("1",),), ((1, 1), (1, [2])), ((1, 1), (1, "[2]"))]
+ILLEGAL_VALUE_MATRIX_LIST = [((), (1,)), ((1,), (1,)), ((1, 2), (1,))]
+VERY_BIG_MATRIX = (((1, ) * 100), ) * 100
+
+
+@pytest.fixture
+def matrix_factory():
+    def _matrix_factory(matrix):
+        return Matrix(matrix)
+
+    return _matrix_factory
+
+
+@pytest.fixture
+def compare_matrix_rows():
+    def _compare_matrix_rows(matrix_a, matrix_b):
+        for row_a, row_b in zip(matrix_a, matrix_b):
+            assert row_a == row_b
+
+    return _compare_matrix_rows
+
+
+########################################################################################################################
+@pytest.mark.parametrize('expected_matrix, expected_exception',
+                         list(zip(LEGAL_MATRIX_LIST, [None] * len(LEGAL_MATRIX_LIST))) +
+                         list(zip(ILLEGAL_TYPE_MATRIX_LIST, [TypeError] * len(ILLEGAL_TYPE_MATRIX_LIST))) +
+                         list(zip(ILLEGAL_TYPE_MATRIX_CELL, [TypeError] * len(ILLEGAL_TYPE_MATRIX_CELL))) +
+                         list(zip(ILLEGAL_VALUE_MATRIX_LIST, [ValueError] * len(ILLEGAL_VALUE_MATRIX_LIST))))
+def test_matrix(matrix_factory, expected_matrix, expected_exception):
+    if expected_exception is None:
+        assert matrix_factory(expected_matrix).matrix == expected_matrix
+    else:
+        with pytest.raises(expected_exception):
+            matrix_factory(expected_matrix)
+
+
+########################################################################################################################
+@pytest.mark.parametrize("test_input, expected_exception",
+                         list(zip(LEGAL_MATRIX_LIST, [None] * len(LEGAL_MATRIX_LIST))) +
+                         list(zip(ILLEGAL_TYPE_MATRIX_LIST, [TypeError] * len(ILLEGAL_TYPE_MATRIX_LIST))) +
+                         list(zip(ILLEGAL_TYPE_MATRIX_CELL, [TypeError] * len(ILLEGAL_TYPE_MATRIX_CELL))) +
+                         list(zip(ILLEGAL_VALUE_MATRIX_LIST, [ValueError] * len(ILLEGAL_VALUE_MATRIX_LIST))))
+def test_is_valid_matrix(test_input, expected_exception):
+    if expected_exception is None:
+        Matrix.is_valid_matrix(test_input)
+    else:
+        with pytest.raises(expected_exception):
+            Matrix.is_valid_matrix(test_input)
+
+
+########################################################################################################################
+@pytest.mark.parametrize('expected_matrix', LEGAL_MATRIX_LIST)
```

Comment:
The whole constants for each case is confusing, i want to see what are the values you are testing.